Instagram Reels Bot/Helpers/IGComponentBuilder.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using Discord;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Instagram_Reels_Bot.Helpers
+{
+    internal class IGComponentBuilder
+    {
+        /// <summary>
+        /// The Response from the Instagram processor.
+        /// </summary>
+        private InstagramProcessorResponse Response;
+        private string Requester;
+        private bool RequesterIsKnown
+        {
+            get
+            {
+                return !string.IsNullOrWhiteSpace(Requester);
+            }
+        }
+        /// <summary>
+        /// Create an instance of the component builder.
+        /// </summary>
+        /// <param name="response"></param>
+		public IGComponentBuilder(InstagramProcessorResponse response, string requester)
+        {
+            this.Response = response;
+            this.Requester = requester;
+        }
+        /// <summary>
+        /// For use when requester is not needed or unknown.
+        /// </summary>
+        /// <param name="response"></param>
+		public IGComponentBuilder(InstagramProcessorResponse response)
+        {
+            this.Response = response;
+        }
+        /// <summary>
+        /// Automatically determines what component type to use
+        /// </summary>
+        /// <returns>An message component</returns>
+		public MessageComponent AutoSelector()
+        {
+            if (Response.onlyAccountData)
+            {
+                return AccountComponent(); // Link in bio
+            }
+            return PostComponent(); // View on IG
+        }
+        /// <summary>
+        /// The basic structure for Component
+        /// </summary>
+        /// <returns>An component builder with basic settings</returns>
+		public ComponentBuilder BaseComponent()
+        {
+            ComponentBuilder component = new ComponentBuilder();
+
+            return component;
+        }
+        /// <summary>
+        /// Builds a Component for IG posts
+        /// </summary>
+        /// <returns></returns>
+		public MessageComponent PostComponent()
+        {
+            var component = BaseComponent();
+
+            // create button
+            ButtonBuilder button = new ButtonBuilder();
+            button.Label = "View on IG";
+            button.Style = ButtonStyle.Link;
+            button.Url = Response.postURL.ToString();
+
+            // add button to component
+            component.WithButton(button);
+
+            return component.Build();
+        }
+        /// <summary>
+        /// Builds a Component for accounts (profile)
+        /// </summary>
+        /// <returns></returns>
+		public MessageComponent AccountComponent()
+        {
+            var component = BaseComponent();
+
+            if (Response.externalURL == null || String.IsNullOrEmpty(Response.externalURL.ToString())) // if there is no external url, then no button

```

Comment:
This can also be simplified to just `String.IsNullOrEmpty(Response.externalURL.ToString()`. `Response.externalURL == null` is redundant.

Reply:
can i just use `Response.externalURL == null`?. because when i use `String.IsNullOrEmpty(Response.externalURL.ToString()` it throws an error:
```
  System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Instagram Reels Bot
  StackTrace:
   at Instagram_Reels_Bot.Helpers.IGComponentBuilder.AccountComponent() in D:\IGEmbed\git\Instagram Reels Bot\Helpers\IGComponentBuilder.cs:line 73
```

Reply:
Odd. it should check for that. Ill have to see why not.

Reply:
Oh wait, thats because the URI is null and the .ToString() is throwing it. Makes sense now. Keep the == null.

Reply:
You probably dont need the String.IsNullOrEmpty() then as URI class _probably_ wont allow for empty URIs aside from null, but ill have to look into that.